python/dolfinx/fem/function.py

```diff
@@ -330,10 +359,16 @@ def _interpolate(u, cells):
         def _(u_ptr, cells):
             self._cpp_object.interpolate_ptr(u_ptr, cells)
 
+        @_interpolate.register(Expression)
+        def _(expr: Expression, cells: np.ndarray = None):
```

Comment:
Type hint is wrong `None` is not a `np.ndarray`.

Could remove type hint since this is a 'private' function.